back-end/java/main/java/src/DormRoom/DormRoom.java

```diff
@@ -0,0 +1,157 @@
+package src.DormRoom;
+
+/**
+ * The {@code DormRoom} class represents a dormitory room with various attributes such as size,
+ * capacity, building location, and amenities. It is designed to encapsulate data about a single
+ * dorm room, including its building, the type of bathroom it has, and whether it includes a kitchen
+ * or is part of a suite. Instances of this class are immutable once constructed.
+ *
+ * <p>Usage example:
+ * <pre>
+ * DormRoom myRoom = new DormRoom(200, 255, 2,
+ *    'example.com', true, false,
+ *    BathroomType.Private, "Barbour Hall");
+ * </pre>
+ */
+public class DormRoom {
```

Comment:
Based on our discussion earlier on how suites should be displayed on the frontend, I recommend that we also make a "suite" class, that maybe has a list of dorm rooms, a haskitcken, hasbathroom, TOTAL capacity, floor, and common area size fields. Maybe make an interface or abstract class that a dorm room and suite both implement